services/meta/client.go

```diff
@@ -343,24 +343,24 @@ func (c *Client) CreateDatabaseWithRetentionPolicy(name string, rpi *RetentionPo
 		return nil, ErrRetentionPolicyDurationTooLow
 	}
 
-	if _, err := c.CreateDatabase(name); err != nil {
-		return nil, err
-	}
-
-	if err := c.DropRetentionPolicy(name, rpi.Name); err != nil {
-		return nil, err
+	if db, _ := c.Database(name); db != nil {
+		// Check if the retention policy already exists. If it does and matches
+		// the desired retention policy, exit with no error.
+		if rp := db.RetentionPolicy(rpi.Name); rp != nil {
+			if rp.ReplicaN != rpi.ReplicaN || rp.Duration != rpi.Duration {
+				return nil, ErrRetentionPolicyExists
```

Comment:
We would have to likely add a new error. That error is thrown if you try to use `CREATE RETENTION POLICY` which makes sense as the text. What do you propose making this error into?